deno/lib/types.ts

```diff
@@ -4726,8 +4735,13 @@ export type BRAND<T extends string | number | symbol> = {
 
 export class ZodBranded<
   T extends ZodTypeAny,
-  B extends string | number | symbol
-> extends ZodType<T["_output"] & BRAND<B>, ZodBrandedDef<T>, T["_input"]> {
+  B extends string | number | symbol,
+  S extends boolean
```

Comment:
this could be 

```suggestion
  S extends boolean = false
```

if we don't want breaking changes